### PR TITLE
[Update] Token Rework

### PR DESCRIPTION
**Pull request by kbrizzle**
Reworks the `Token` type to remove dynamic `.decimals()` normalization as this pattern has the potential to be more confusing to users than it is helpful to developers.

- Renames `Token` to `TokenOrEther18`
  - Removes all dynamic decimal normalization from `TokenOrEther18`, maintains the `ether` support.
- Removes `.decimals()` getters from each `Token` type, maintains `.name()` and `.symbol()` getters.
- Adds `ZERO` constant to each `Token` type.
- Adds `.isZero()` method to each `Token` type.
- Adds `.eq()` comparator to each `Token` type.
- Bumps root version to `v0.1.3`.

### CODE REVIEW

File: contracts/token/types/TokenOrEther18.sol [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+ * @dev Normalizes token operations with Ether operations (using a magic Ether address)
+ */
+library TokenOrEther18Lib {
+    using Address for address;
+    using SafeERC20 for IERC20;
+
+    error TokenOrEther18PullEtherError();
+    error TokenOrEther18ApproveEtherError();
+
+    TokenOrEther18 public constant ZERO = TokenOrEther18.wrap(address(0));
```
Comment (arjun-io) [inline contracts/token/types/TokenOrEther18.sol:31]: Why do we need a ZERO value?

**Review by arjun-io — APPROVED** [review]
👍

File: contracts/token/types/TokenOrEther18.sol (hunk 1 above)
Comment (kbrizzle) [reply contracts/token/types/TokenOrEther18.sol:31]: I added `ZERO` to match the defaults we have in the `Fixed` types.

It's probably not going to be super widely used, but there's some cases where it'd be nice to have like populating defaults in structs or validation checks.

**Comment by kbrizzle** [comment]
Added another `isZero()` function here as well, could you check that quick @arjun-io

**Review by arjun-io — APPROVED** [review]

**Pull request merged**